scyllacloud/hostdialer_test.go

```diff
@@ -0,0 +1,400 @@
+package scyllacloud
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+const (
+	testTimeout = time.Second
+)
+
+func TestHostSNIDialer_InvalidConnectionConfig(t *testing.T) {
```

Comment:
why not make it in one function when already doing table based tests?

Reply:
tests are separated because they are testing distinct things. One validates error handling, second one the expected behavior. Error handling doesn't require working TLS server not SNI capturing

Reply:
this feels weird https://github.com/scylladb/gocql/pull/106/files#diff-1abd0daf160a37ead702fdfa4d5132eece4da32ffc0be0f897515b83d445b7a2R34

Reply:
but I don't want to nit pick on this - marking it as resolved